x-pack/plugins/secops/public/store/local/app/selectors.ts

```diff
@@ -6,12 +6,24 @@
 
 import { createSelector } from 'reselect';
 
+import { keys, memoize } from 'lodash/fp';
+import { Note } from '../../../lib/note';
 import { State } from '../../reducer';
 import { NotesById } from './model';
 
 const selectNotesById = (state: State): NotesById => state.local.app.notesById;
 
-export const notesByIdSelector = createSelector(
-  selectNotesById,
-  notesById => notesById
-);
+const getNotes = (notesById: NotesById, eventIds: string[]) =>
+  keys(notesById).reduce((acc: Note[], noteId: string) => {
+    if (eventIds.includes(noteId)) {
+      const note: Note = notesById[noteId];
+      acc = [...acc, note];
+    }
+    return acc;
+  }, []);
+
+export const notesByIdsSelector = () =>
+  createSelector(
+    selectNotesById,
+    (notesById: NotesById) => memoize((eventIds: string[]): Note[] => getNotes(notesById, eventIds))
```

Comment:
I recommend using https://github.com/alexreardon/memoize-one here (it's already a dependency of the project)

Reply:
Yes that's correct